### ISSUE TITLE
Validate a query parameter against one of multiple subSchemas

### ISSUE THREAD

**Issue by Mouad-B** [kogosoftwarellc/open-api#535]
I want to specify multiple possible subschemas for query param but no error is thrown when any of them matches to query passed as a parameter. Here is an example:

```
openapi: '3.0.0'
paths:
  /ressource
    get:
      parameters:
        - $ref: '#/components/parameters/user'
components:
  parameters:
    user:
      in: query
      name: user
      required: true
      schema:
        oneOf:
          - pattern: '^[a-z]{4}$'
          - enum: [os1, os2]
```

No validation is done against the two possible query formats. express-openapi fires however an error when no user query is passed but doesn't inspect its value.. Am I missing something here?
Thanks for your help!

**Comment by jsdevel** [comment]
is that valid openapi?

**Comment by Mouad-B** [comment]
yes, I guess so

**Comment by jsdevel** [comment]
@Mouad-B can you find out?

**Comment by Mouad-B** [comment]
@jsdevel not really, can you help?

**Comment by jsdevel** [comment]
@Mouad-B sorry, but i don't have the time.  i mainly review PRs and publish changes when the library is ready

**Comment by Mouad-B** [comment]
@jsdevel I thought you found a mistake in my yaml but after verification the one I gave is totally correct. I opened this issue because there is an issue with express-openapi which doesn't validate a query parameter against multiple sachems. You can reproduce this bug with the example I gave.
Don't hesitate if you need further information

**Comment by jsdevel** [comment]
please open a pr

**Comment by keijokapp** [comment]
This (and #369) seems to be result of [whitelisting](https://github.com/kogosoftwarellc/open-api/blob/master/packages/openapi-jsonschema-parameters/index.ts#L72-L77) schema properties. Is there a reason why whitelisting is necessary?

**Comment by jsdevel** [comment]
@keijokapp feel free to open a pr

**Comment by keijokapp** [comment]
Sorry, I've been crazy busy.

I guess library tries to keep compatibility with pre-OpenAPI-3.x, though I haven't checked [2.x](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md) spec for compliance.

What's current position in regards to pre-3.x compatibility? The current concept seems to be quite a mess and could be greatly simplified - i.e. removing support for validation keywords directly in Parameter Object.

I would submit PR asap but would like to know if I should dive into 2.0 spec and try to keep compatibility, which I guess won't be straightforward.

**Comment by jsdevel** [comment]
> I would submit PR asap but would like to know if I should dive into 2.0 spec and try to keep compatibility, which I guess won't be straightforward.

we are striving to be 2.0 compliant.  having said that, we rely on test coverage to ensure that we haven't broken anything, so fly free!

**Comment by morriskimani** [comment]
Hello @jsdevel. Seems like this was fixed by #621 but it was never published as mentioned in this issue #662. Would love to know if there are any plans to cut a release that has these changes. Thanks

**Comment by jsdevel** [comment]
7.0.0 is out.  first release/publish using lerna!  please lmk if it's all working correctly.  now that we have lerna setup, publishes should happen a lot faster.  the old process was too tedious for me.  now it's joyful :joy:
